homeassistant/components/sensor/websocket_api.py

```diff
@@ -0,0 +1,33 @@
+"""The sensor websocket API."""
+from __future__ import annotations
+
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.components import websocket_api
+from homeassistant.core import HomeAssistant, callback
+
+from .const import DEVICE_CLASS_UNITS
+
+
+@callback
+def async_setup(hass: HomeAssistant) -> None:
+    """Set up the sensor websocket API."""
+    websocket_api.async_register_command(hass, ws_device_class_units)
+
+
+@callback
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "sensor/device_class_units",
+        vol.Required("device_class"): str,
```

Comment:
```suggestion
        vol.Required("device_class"): vol.In(DEVICE_CLASS_UNITS),
```

Reply:
Some devices can have custom `device_class`. With this change, front-end need to hardcode authorized device_classes list to avoid error.

Reply:
Yeah, I agree this is not a good change.